hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -17,210 +17,260 @@
 
 from __future__ import absolute_import, division, print_function
 
-from hypothesis.internal.compat import hbytes, hrange
-from hypothesis.internal.conjecture.data import Status
+import attr
+
+from hypothesis.errors import Flaky, HypothesisException
+from hypothesis.internal.compat import hbytes, int_to_bytes
+from hypothesis.internal.conjecture.data import (
+    ConjectureData,
+    DataObserver,
+    Status,
+    StopTest,
+    bits_to_bytes,
+)
+from hypothesis.internal.conjecture.junkdrawer import IntList, uniform
+
+
+class PreviouslyUnseenBehaviour(HypothesisException):
```

Comment:
I'd define this in `hypothesis.errors`, so you can find all our exception types in one file, and import it here.

Reply:
I kinda feel like `hypothesis.errors` is tacitly public API and this is an exception purely internal to conjecture. It really should never escape into the wider world, so having it be local seems reasonable. This is also how we do `RunIsComplete`